crates/amaru-consensus/src/consensus/tx_submission/tests/node.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright 2025 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::consensus::tx_submission::tests::{
+    MockTransport, SizedMempool, TxSubmissionClient, TxSubmissionServer,
+};
+use crate::consensus::tx_submission::{Blocking, ServerParams};
+use amaru_kernel::peer::Peer;
+use amaru_mempool::strategies::InMemoryMempool;
+use amaru_ouroboros_traits::can_validate_transactions::mock::MockCanValidateTransactions;
+use amaru_ouroboros_traits::{CanValidateTransactions, Mempool};
+use pallas_network::miniprotocols::txsubmission::{EraTxBody, EraTxId, Message};
+use pallas_primitives::conway::Tx;
+use std::sync::Arc;
+use tokio::sync::mpsc::Receiver;
+use tokio::task::JoinHandle;
+
+/// A test node consisting of a TxSubmissionClient and TxSubmissionServer,
+/// their respective mempools, connected via a MockTransport.
+pub struct Node {
```

Comment:
This isn’t really a node, it is parts of two nodes, correct? (the client and the server)